tests/DotNetBoilerplate.Tests.Integration/BookStoreEndpointsTests.cs

```diff
@@ -0,0 +1,248 @@
+            Description = "Description"
+        };
+
+        //Act
+        var response = await testsFixture.Client.PostAsJsonAsync("book-stores", request);
+
+        //Assert
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+
+        bookStore.ShouldNotBeNull();
+        bookStore!.Name.ShouldBe(request.Name);
+    }
+
+    [Fact]
+    public async Task GivenBookStoreExists_CreateBookStore_ShouldFail()
+    {
+        //Arrange
+        var existingBookStore = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+        var request = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore2",
+            Description = "Description2"
+        };
+
+        //Act
+        var firstResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", existingBookStore);
+        var secondResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", request);
+
+        //Assert
+        firstResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+        secondResponse.StatusCode.ShouldBe(HttpStatusCode.BadRequest);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStores = await dbContext.BookStores.ToListAsync();
+        bookStores.Count.ShouldBe(1);
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+        bookStore.ShouldNotBeNull();
+        bookStore!.Name.ShouldBe(existingBookStore.Name);
+    }
+
+    public async Task GivenBookStoreExists_UpdateBookStore_ShouldSucceed()
+    {
+        //Arrange
+        var createRequest = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+        var createResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", createRequest);
+        createResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+        bookStore.ShouldNotBeNull();
+        bookStore!.Name.ShouldBe(createRequest.Name);
+
+
+        var updateRequest = new UpdateBookStoreEndpoint.Request
+        {
+            Name = "Updated book store",
+            Description = "Updated description"
+        };
+
+        //Act
+        var updateResponse = await testsFixture.Client.PutAsJsonAsync($"book-stores/{bookStore.Id}", updateRequest);
```

Comment:
Tutaj możliwe że chcieliście wyciągnąć to z bazy żebt mieć id, na szczęście my sobie zwracamy Id stworzonego zasobu w response więc można sobie użyć  var result = await response.Content.ReadFromJsonAsync<CreateBookStoreEndpoint.Response>(); i wtedy mamy dostęp do tego responsa z endpointa create który nam zwraca Id stworzonego book stora.

Możliwość łatwego wyciągnięcia DB context i robienie na nim dużo jest kuszące w testach integracyjnych, ale jak możemy do czegoś wykorzystać istniejące endpointy to zachęcam do tego. Abstrakcyjna może sytuacja ale może się baza potem zmienić czy cos więc wtedy jak w mniejszej ilości miejsc mamy db context używant tym szybciej przeniesiemy